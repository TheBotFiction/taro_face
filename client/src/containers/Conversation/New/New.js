import React, { Component } from 'react'
// import PropTypes from 'prop-types'

export class ConversationNewContainer extends Component {
  static propTypes = {

  }

  render() {
    return (
      <div>

      </div>
    )
  }
}

export default ConversationNewContainer
