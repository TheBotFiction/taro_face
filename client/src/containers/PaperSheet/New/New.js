/**
 * @flow
 */

import React, { Component, Fragment } from 'react'
import type { Node } from 'react'
import type { TermType } from 'types'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import NewComponent from 'components/PaperSheet/New'
import PreviewContainer from './Preview'

type Props = {}
type State = {
  selectedTermId: null | number
}

type QueryResult = {
  loading: boolean,
  data: {
    terms: Array<TermType> | void
  }
}

const INDEX_TERM_QUERY = gql`
  query indexTerms {
    terms {
      id
      term
    }
  }
`

const loadOptions = (suggestions) => {
  return (inputValue, callback) => {
    setTimeout(() => {
      const matchedSuggestions: Array<Object> = suggestions.filter((option): boolean => (
        option.label.toLowerCase().indexOf(inputValue) >= 0
      ))
      callback(matchedSuggestions)
    })
  }
}

class NewContainer extends Component<Props, State> {
  onSelectTerm: Function

  constructor (props) {
    super(props)

    this.onSelectTerm = this.onSelectTerm.bind(this)
  }

  state = {
    selectedTermId: null
  }

  onSelectTerm (termId: number): void {
    this.setState({selectedTermId: termId})
  }

  render () {
    const { selectedTermId } = this.state
    const previewSlot: Node = <PreviewContainer termId={selectedTermId} />

    return(
      <Query query={INDEX_TERM_QUERY}>
        {this.renderQueriedNewComponent}
      </Query>
    )
  }

  renderQueriedNewComponent ({loading, data}: QueryResult): Node {
    if (loading) return <h2>Loading ...</h2>
    if (!data) return null

    const { selectedTermId } = this.state
    const previewSlot: Node = <PreviewContainer termId={selectedTermId} />

    let suggestions: Array<Object> = []
    if (data.terms) {
      suggestions = data.terms.map(term => ({value: term.id, label: term.term}))
    }
    return (
      <Fragment>
        <NewComponent
          loadOptions={loadOptions(suggestions)}
          onSelectTerm={this.onSelectTerm}
          previewSlot={previewSlot}
        />
      </Fragment>
    )
  }
}

export default NewContainer