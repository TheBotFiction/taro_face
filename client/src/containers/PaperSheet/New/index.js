import PaperSheetNewContainer from './New'
export default PaperSheetNewContainer