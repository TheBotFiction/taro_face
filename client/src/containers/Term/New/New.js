import React, { Component } from 'react'
import PropTypes from 'prop-types'

export class TermNewContainer extends Component {
  static propTypes = {

  }

  render() {
    return (
      <div>
        
      </div>
    )
  }
}

export default TermNewContainer
