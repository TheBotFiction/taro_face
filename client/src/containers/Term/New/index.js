import TermCreateContainer from './New';

export default TermCreateContainer;
