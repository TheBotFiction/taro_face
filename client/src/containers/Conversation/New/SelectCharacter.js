/**
 * @flow
 */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import gql from 'graphql-tag'
import { Query } from 'react-apollo'
import { SelectCharacterComponent } from 'components/Conversation/New'

export class SelectCharacterContainer extends Component<*, *> {
  static propTypes = {
    loading: PropTypes.bool.isRequired,
    data: PropTypes.object
  }

  render () {
    const { data, loading }: { data: Object, loading: boolean } = this.props
    if (loading) return 'Loading...'
    if (!data) return null
    const { characters } = data
    return (
      <SelectCharacterComponent {...this.props} data={characters} />
    )
  }
}

const INDEX_CHARACTERS = gql`
  query indexCharacters {
    characters {
      id
      name
    }
  }
`

export default (props: any) => {
  return (
    <Query query={INDEX_CHARACTERS}>
      {({loading, data}) => (
        <SelectCharacterContainer
          {...props}
          loading={loading}
          data={data}
        />
      )}
    </Query>
  )
}
