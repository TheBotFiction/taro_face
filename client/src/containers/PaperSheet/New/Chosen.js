/**
 * @flow
 */
import React, { Component } from 'react'
import { ChosenComponent } from 'components/PaperSheet/New'

type Props = *

class ChosenContainer extends Component<Props, *> {
  render () {
    return <ChosenComponent />
  }
}

export default ChosenContainer