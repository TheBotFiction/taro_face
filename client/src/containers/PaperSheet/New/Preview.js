import React, { Component } from 'react'
import type { Node } from 'react'
import _ from 'lodash'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import { PreviewComponent } from 'components/PaperSheet/New'

type QueryResult = {
  loading: boolean,
  data: {
    term: TermType
  } | void
}

const SHOW_TERM_QUERY = gql`
  query showTerm($id: ID!) {
    term(id: $id) {
      id
      term
      reading
      meaning
      samplePhrases(limit: 2) {
        id
        phrase
      }
      similars {
        id
        term
      }
    }
  }
`

class PreviewContainer extends Component {
  chooseQuestion: Function
  renderQueriedPreviewComponent: Function

  constructor (props) {
    super(props)
    this.chooseQuestion = this.chooseQuestion.bind(this)
    this.renderQueriedPreviewComponent = this.renderQueriedPreviewComponent.bind(this)
  }

  // getDerivedStateFromProps (props, state): * {
  //   const { term }: { term: TermType } = props
  //   const samplePhrase: SamplePhraseType | void = _.sample(term.samplePhrases)
  //   return {
  //     term: {
  //       id: term.id,
  //       term: term.term
  //     },
  //     similars: term.similars,
  //     samplePhrase
  //   }
  // }

  chooseQuestion (payload): void {
    console.log(payload)
  }

  render () {
    const { termId } = this.props
    if (!termId) return null

    return (
      <Query query={SHOW_TERM_QUERY} variables={{ id: termId }}>
        { this.renderQueriedPreviewComponent }
      </Query>
    )
  }

  renderQueriedPreviewComponent ({loading, data}: QueryResult): Node {
    if (loading) return <h2>Loading ...</h2>
    if (!data) return null

    const { term }: { term: TermType } = data
    const samplePhrase: SamplePhraseType | void = _.sample(term.samplePhrases)

    if (!samplePhrase) return null

    const normalizedTerm: TermType = { id: term.id, term: term.term }
    const similars: Array<TermType> = term.similars
    const answers: Array<TermType> = _.shuffle([ term, ...similars ])

    return (
      <PreviewComponent
        term={normalizedTerm}
        samplePhrase={samplePhrase}
        answers={answers}
        chooseQuestion={this.chooseQuestion}
      />
    )
  }
}

export default PreviewContainer