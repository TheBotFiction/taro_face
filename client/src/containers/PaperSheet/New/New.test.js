import React from 'react'
import TestRenderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { MockedProvider } from 'react-apollo/test-utils'
import wait from 'waait'
import * as queries from 'constants/queries'

import New from './New'

const mocks = [
  {
    request: {
      query: queries.INDEX_TERM_QUERY
    },
    result: {
      data: {
        terms: [{
          id: 1,
          term: 'Fake term 1',
          meaning: 'Fake meaning 1',
          reading: 'Fake reading 1',
          __typename: 'TermType'
        }]
      }
    }
  },
  {
    request: {
      query: queries.CREATE_PAPER_SHEET_MUTATION,
      variables: {
        questions: [
          {
            term: 'Fake term 2',
            phrase: 'Fake phrase 2',
            answers: [
              'Fake term 1',
              'Fake term 2',
              'Fake term 3',
              'Fake term 4'
            ]
          }
        ]
      }
    }
  }
]

describe('component Term/New', () => {
  it('renders without crash', () => {
    shallow(
      <MockedProvider mocks={mocks}>
        <New />
      </MockedProvider>
    )
  })

  it('renders as expected', async () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={mocks}>
        <New />
      </MockedProvider>
    )

    await wait(1000)

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
})