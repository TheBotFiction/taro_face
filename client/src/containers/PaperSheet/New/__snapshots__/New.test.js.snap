// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Term/New renders as expected 1`] = `
Array [
  <header
    className="MuiPaper-root-40 MuiPaper-elevation4-46 MuiAppBar-root-31 MuiAppBar-positionFixed-32 MuiAppBar-colorPrimary-38 mui-fixed Header-appBar-14 Header-transparent-27 Header-fixed-16"
>
    <div
        className="MuiToolbar-root-67 MuiToolbar-regular-69 MuiToolbar-gutters-68 Header-container-17"
    >
        <div
            className="Header-flex-18"
        >
            <MockedButtonBase
                className="MuiButton-root-71 MuiButton-text-73 MuiButton-flat-76 Header-title-19"
                component="button"
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="MuiButton-focusVisible-90"
                type="button"
            >
                <span
                    className="MuiButton-label-72"
                >
                    Taro's Face
                </span>
            </MockedButtonBase>
        </div>
        <div
            className="HiddenCss-smDown-102"
        >
            <ul
                className="MuiList-root-124 MuiList-padding-125 HeaderLinks-list-112"
            >
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 HeaderLinks-listItem-113"
                    disabled={false}
                >
                    <div>
                        <div>
                            <MockedButtonBase
                                aria-haspopup="true"
                                aria-label="Notifications"
                                aria-owns={null}
                                buttonRef={[Function]}
                                className="MuiButton-root-71 MuiButton-text-73 MuiButton-flat-76 RegularButton-button-159 RegularButton-transparent-173 HeaderLinks-navLink-115"
                                component="button"
                                disabled={false}
                                focusRipple={true}
                                focusVisibleClassName="MuiButton-focusVisible-90"
                                onClick={[Function]}
                                type="button"
                            >
                                <span
                                    className="MuiButton-label-72"
                                >
                                    <svg
                                        aria-hidden="true"
                                        className="MuiSvgIcon-root-181 CustomDropdown-buttonIcon-153"
                                        color={undefined}
                                        focusable="false"
                                        role="presentation"
                                        viewBox="0 0 24 24"
                                    >
                                        <path
                                            d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                                        />
                                        <path
                                            d="M0 0h24v24H0z"
                                            fill="none"
                                        />
                                    </svg>
                                    Learn
                                    <b
                                        className="CustomDropdown-caret-154"
                                    />
                                </span>
                            </MockedButtonBase>
                        </div>
                    </div>
                </li>
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 HeaderLinks-listItem-113"
                    disabled={false}
                >
                    <MockedButtonBase
                        className="MuiButton-root-71 MuiButton-text-73 MuiButton-flat-76 RegularButton-button-159 RegularButton-transparent-173 HeaderLinks-navLink-115"
                        component="button"
                        disabled={false}
                        focusRipple={true}
                        focusVisibleClassName="MuiButton-focusVisible-90"
                        href="#"
                        target="_blank"
                        type="button"
                    >
                        <span
                            className="MuiButton-label-72"
                        >
                            <svg
                                aria-hidden="true"
                                className="MuiSvgIcon-root-181 HeaderLinks-icons-119"
                                color={undefined}
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                            >
                                <path
                                    d="M0 0h24v24H0z"
                                    fill="none"
                                />
                                <path
                                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
                                />
                            </svg>
                             Take Exam
                        </span>
                    </MockedButtonBase>
                </li>
            </ul>
        </div>
    </div>
    <div
        className="HiddenCss-mdUp-104"
    />
</header>,
  <div
    className="Parallax-parallax-203"
    style={
        Object {
            "backgroundImage": "url(bg.jpg)",
            "transform": "translate3d(0,0px,0)",
          }
    }
>
    <div
        className="Layout-container-6"
    >
        <div
            className="MuiGrid-container-206"
        >
            <div
                className="MuiGrid-item-207"
            >
                <div
                    className="Layout-brand-7"
                >
                    <h1
                        className="Layout-title-8"
                    >
                        New Paper Sheet
                    </h1>
                    <h3
                        className="Layout-subtitle-9"
                    />
                </div>
            </div>
        </div>
    </div>
</div>,
  <div
    className="Layout-main-10 Layout-mainRaised-11"
>
    <main
        className="PaperSheetNewComponent-root-1"
    >
        <div
            className="MuiGrid-container-206 PaperSheetNewComponent-termSelectZone-5"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-8-242 MuiGrid-grid-lg-8-284"
            >
                <div
                    className="css-10nd86i"
                    id={undefined}
                    onKeyDown={[Function]}
                >
                    <div
                        aria-describedby={undefined}
                        className="MuiFormControl-root-304 MuiFormControl-fullWidth-307"
                    >
                        <div
                            className="MuiInputBase-root-321 MuiInput-root-308 MuiInput-underline-312 MuiInputBase-fullWidth-330 MuiInput-fullWidth-315 MuiInputBase-formControl-322 MuiInput-formControl-309"
                            onClick={[Function]}
                        >
                            <div
                                aria-invalid={false}
                                autoComplete={undefined}
                                autoFocus={undefined}
                                className="MuiInputBase-input-331 MuiInput-input-316 Control-input-303"
                                defaultValue={undefined}
                                disabled={false}
                                id={undefined}
                                name={undefined}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyDown={undefined}
                                onKeyUp={undefined}
                                onMouseDown={[Function]}
                                onTouchEnd={[Function]}
                                placeholder={undefined}
                                readOnly={undefined}
                                required={false}
                                rows={undefined}
                                type="text"
                                value={undefined}
                            >
                                <div
                                    className="ValueContainer-valueContainer-338"
                                >
                                    <p
                                        className="MuiTypography-root-340 MuiTypography-body1-349 MuiTypography-colorTextSecondary-373 Placeholder-placeholder-339"
                                    >
                                        Search a term
                                    </p>
                                    <div
                                        className="css-1hrkh7z"
                                    >
                                        <div
                                            className=""
                                            style={
                                                Object {
                                                    "display": "inline-block",
                                                  }
                                            }
                                        >
                                            <input
                                                aria-autocomplete="list"
                                                aria-label={undefined}
                                                aria-labelledby={undefined}
                                                autoCapitalize="none"
                                                autoComplete="off"
                                                autoCorrect="off"
                                                className={undefined}
                                                disabled={false}
                                                id="react-select-2-input"
                                                onBlur={[Function]}
                                                onChange={[Function]}
                                                onFocus={[Function]}
                                                spellCheck="false"
                                                style={
                                                    Object {
                                                        "background": 0,
                                                        "border": 0,
                                                        "boxSizing": "content-box",
                                                        "color": "inherit",
                                                        "fontSize": "inherit",
                                                        "opacity": 1,
                                                        "outline": 0,
                                                        "padding": 0,
                                                        "width": "1px",
                                                      }
                                                }
                                                tabIndex="0"
                                                type="text"
                                                value=""
                                            />
                                            <div
                                                style={
                                                    Object {
                                                        "height": 0,
                                                        "left": 0,
                                                        "overflow": "scroll",
                                                        "position": "absolute",
                                                        "top": 0,
                                                        "visibility": "hidden",
                                                        "whiteSpace": "pre",
                                                      }
                                                }
                                            >
                                                
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div
                                    className="css-1wy0on6"
                                >
                                    <span
                                        className="css-d8oujb"
                                    />
                                    <div
                                        aria-hidden="true"
                                        className="css-1ep9fjw"
                                        onMouseDown={[Function]}
                                        onTouchEnd={[Function]}
                                    >
                                        <svg
                                            aria-hidden="true"
                                            className="css-19bqh2r"
                                            focusable="false"
                                            height={20}
                                            viewBox="0 0 20 20"
                                            width={20}
                                        >
                                            <path
                                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                            />
                                        </svg>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-4-238 MuiGrid-grid-lg-4-280 PaperSheetNewComponent-buttonContainer-2"
            >
                <MockedButtonBase
                    className="MuiButton-root-71 MuiButton-text-73 MuiButton-textPrimary-74 MuiButton-outlined-79 MuiButton-outlinedPrimary-80 MuiButton-sizeSmall-94 MuiButton-disabled-91"
                    component="button"
                    disabled={true}
                    focusRipple={true}
                    focusVisibleClassName="MuiButton-focusVisible-90"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        className="MuiButton-label-72"
                    >
                        Add To PaperSheet
                        <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-181 PaperSheetNewComponent-rightIcon-3"
                            color={undefined}
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                        >
                            <path
                                d="M0 0h24v24H0V0z"
                                fill="none"
                            />
                            <path
                                d="M21 3H3c-1.11 0-2 .89-2 2v12c0 1.1.89 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.11-.9-2-2-2zm0 14H3V5h18v12zm-5-7v2h-3v3h-2v-3H8v-2h3V7h2v3h3z"
                            />
                        </svg>
                    </span>
                </MockedButtonBase>
            </div>
        </div>
        <hr
            className="MuiDivider-root-375 PaperSheetNewComponent-divider-4"
        />
        <div
            className="MuiGrid-container-206"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-8-242 MuiGrid-grid-lg-8-284"
            />
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-4-238 MuiGrid-grid-lg-4-280 PaperSheetNewComponent-buttonContainer-2"
            >
                <MockedButtonBase
                    className="MuiButton-root-71 MuiButton-contained-82 MuiButton-containedSecondary-84 MuiButton-raised-85 MuiButton-raisedSecondary-87 MuiButton-sizeSmall-94 MuiButton-disabled-91"
                    component="button"
                    disabled={true}
                    focusRipple={true}
                    focusVisibleClassName="MuiButton-focusVisible-90"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        className="MuiButton-label-72"
                    >
                        Create PaperSheet
                        <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-181 PaperSheetNewComponent-rightIcon-3"
                            color={undefined}
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                        >
                            <path
                                d="M0 0h24v24H0z"
                                fill="none"
                            />
                            <path
                                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                            />
                        </svg>
                    </span>
                </MockedButtonBase>
            </div>
        </div>
    </main>
</div>,
  <div
    align="center"
    className="MuiGrid-container-206 MuiGrid-justify-xs-center-223 AdditionalInfo-root-379"
>
    <div
        className="MuiGrid-item-207 MuiGrid-grid-xs-12-246"
    >
        <h1
            className="MuiTypography-root-340 MuiTypography-headline-345 MuiTypography-alignCenter-363"
        >
            Frequently Asked Questions
        </h1>
    </div>
    <div
        className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-12-246 AdditionalInfo-content-380"
    >
        <div
            className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-12-246 MuiGrid-grid-sm-6-254"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-1-235"
            />
            <div
                className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-11-245"
            >
                <div
                    className="MuiGrid-item-207 MuiGrid-grid-xs-12-246"
                >
                    <h3
                        className="MuiTypography-root-340 MuiTypography-subheading-347"
                    >
                        How to add question to sheet?
                    </h3>
                </div>
            </div>
        </div>
        <div
            className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-12-246 MuiGrid-grid-sm-6-254"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-1-235"
            />
            <div
                className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-11-245"
            >
                <div
                    className="MuiGrid-item-207 MuiGrid-grid-xs-12-246"
                >
                    <h3
                        className="MuiTypography-root-340 MuiTypography-subheading-347"
                    >
                        How to finish the sheet?
                    </h3>
                </div>
            </div>
        </div>
        <div
            className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-12-246 MuiGrid-grid-sm-6-254"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-1-235"
            />
            <div
                className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-11-245"
            >
                <div
                    className="MuiGrid-item-207 MuiGrid-grid-xs-12-246"
                >
                    <h3
                        className="MuiTypography-root-340 MuiTypography-subheading-347"
                    >
                        How will the sheet look like?
                    </h3>
                </div>
            </div>
        </div>
        <div
            className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-12-246 MuiGrid-grid-sm-6-254"
        >
            <div
                className="MuiGrid-item-207 MuiGrid-grid-xs-1-235"
            />
            <div
                className="MuiGrid-container-206 MuiGrid-item-207 MuiGrid-grid-xs-11-245"
            >
                <div
                    className="MuiGrid-item-207 MuiGrid-grid-xs-12-246"
                >
                    <h3
                        className="MuiTypography-root-340 MuiTypography-subheading-347"
                    >
                        How to edit the sheet?
                    </h3>
                </div>
            </div>
        </div>
    </div>
</div>,
  <footer
    className="Footer-footer-384"
>
    <div
        className="Footer-container-387"
    >
        <div
            className="Footer-left-382"
        >
            <ul
                className="MuiList-root-124 MuiList-padding-125 Footer-list-388"
            >
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 Footer-inlineBlock-389"
                    disabled={false}
                >
                    <a
                        className="Footer-block-381"
                        href="#"
                        target="_blank"
                    >
                        Taro's Face
                    </a>
                </li>
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 Footer-inlineBlock-389"
                    disabled={false}
                >
                    <a
                        className="Footer-block-381"
                        href="#"
                        target="_blank"
                    >
                        About us
                    </a>
                </li>
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 Footer-inlineBlock-389"
                    disabled={false}
                >
                    <a
                        className="Footer-block-381"
                        href="#"
                        target="_blank"
                    >
                        Blog
                    </a>
                </li>
                <li
                    className="MuiListItem-root-128 MuiListItem-default-131 MuiListItem-gutters-135 Footer-inlineBlock-389"
                    disabled={false}
                >
                    <a
                        className="Footer-block-381"
                        href="#"
                        target="_blank"
                    >
                        Licenses
                    </a>
                </li>
            </ul>
        </div>
        <div
            className="Footer-right-383"
        >
            © 
            2018
             , made with
             
            <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-181 Footer-icon-390"
                color={undefined}
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
            >
                <path
                    d="M0 0h24v24H0z"
                    fill="none"
                />
                <path
                    d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
            </svg>
             by
             
            <a
                className="Footer-a-385"
                href="#"
                target="_blank"
            >
                Taro's Face
            </a>
             
            for a better learning environment.
        </div>
    </div>
</footer>,
]
`;
